src/coreclr/vm/method.hpp

```diff
@@ -1824,38 +1849,54 @@ class MethodDesc
         m_wFlags |= mdfHasNativeCodeSlot;
     }
 
-    inline bool IsAsyncHelperMethod() const
+    // Historically we use "Async2" to mean methods that can be called via CORINFO_CALLCONV_ASYNCCALL
+    // CONSIDER: We could have a better name for the concept, but it is hard to beat shortness of "Async2"
+    inline bool IsAsync2Method() const
     {
         LIMITED_METHOD_DAC_CONTRACT;
         if (!HasAsyncMethodData())
             return false;
         auto asyncKind = GetAddrOfAsyncMethodData()->kind;
-        return asyncKind == AsyncMethodKind::AsyncThunkHelper ||
-            asyncKind == AsyncMethodKind::AsyncImplHelper;
+        return asyncKind == AsyncMethodKind::Async2VariantThunk ||
+            asyncKind == AsyncMethodKind::Async2VariantImpl ||
+            asyncKind == AsyncMethodKind::Async2ExplicitImpl;
     }
 
-    inline bool IsTaskReturningMethod() const
+    // Is this an Async2-callable variant method?
+    // If yes, the method has another non-Async2 variant.
+    inline bool IsAsync2VariantMethod() const
```

Comment:
Most uses of this, I think, are actually about "is this the variant that was not in metadata?". 
However, in the current IL API it always happens to be the Async2 one, so "is this the variant that is async2?"  is basically the same question.